lib/ansible/modules/packaging/os/package_facts.py

```diff
@@ -139,95 +150,275 @@
           }
         }
 '''
+import json
+
+from abc import ABCMeta, abstractmethod
+
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.basic import AnsibleModule, get_all_subclasses
+from ansible.module_utils.six import with_metaclass
+
+
+class TestFailed(Exception):
+    pass
+
+
+class PkgMgr(with_metaclass(ABCMeta, object)):
```

Comment:
`six.add_metaclass` is better: https://stackoverflow.com/a/37450110/595220.
Also, `__metaclass__ = type` as the top allows us to omit inheriting from the `object` explicitly:
```suggestion
@add_metaclass(ABCMeta)
class PkgMgr:
```

Reply:
Oh, and I'd like you see readable names. both pkg and mgr shortcuts have a lot of different meanings.
`PackageManager` would be more than enough.

Reply:
Ah, add_metaclass  became available when we dropped python-2.4 support (which enabled us to update to a version of six which contained it).  I believe object is needed in the with_metaclass form but not in the add_metaclass form.

Reply:
Yep. And according to stackoverflow it fixes some inheritance quirks.